assembly/src/assembler/mod.rs

```diff
@@ -778,15 +765,15 @@ impl Assembler {
                     // by noop span
                     let else_blk = if else_blk.is_empty() {
                         let basic_block_node = MastNode::new_basic_block(vec![Operation::Noop]);
-                        mast_forest.ensure_node(basic_block_node)
+                        mast_forest.add_node(basic_block_node)
```

Comment:
Would using `MastForest::add_node()` here and in other similar places create a bunch of duplicate nodes in the underlying `MastForest`? Previously, `ensure_node()` checked for duplicates and didn't create a new node if a node with the same digest was already in the forest. But now, that won't happen right?

I was imagining that we'd need a new data structure - something like `MastForestBuilder` which would be use to maintain a temporary map from node hashes to node IDs and endure that we won't create duplicate nodes. At the end we'd call something `MastFoestBuilder::build()` and it would return a `MastForest`.

Another option (and potentially a better one) is to use `AssemblyContext` as the builder. But maybe that's for another PR.

Reply:
Introduced `MastForestBuilder` (for now at least)